quick-start/node-server/app.js

```diff
@@ -0,0 +1,130 @@
+app.use( express.static( 'editor' ) );
+app.get( '/', ( req, res ) => {
+   res.redirect( '/sample/index.html' );
+} );
+
+// Token endpoint
+app.get( '/token', generateToken );
+
+// Create environment
+app.post( '/init', createEnvironment );
+
+app.listen( 3000, () => console.log( `Node-server is listening on port 3000` ) );
+
+
+function generateToken( req, res ) {
+
+   const payload = {
+      aud: environment_id,
+      sub: req.query[ 'user.id' ] || 'user-1',
+      user: {
+         name: req.query[ 'user.name' ] || 'John Doe',
+         avatar: req.query[ 'user.avatar' ] || ''
+      },
+      auth: {
+         'collaboration': {
+            '*': {
+               'role': req.query.role || 'writer'
+            }
+         }
+      }
+   };
+
+   const token = jwt.sign( payload, access_key, { algorithm: 'HS256' } );
+   res.send( token );
+} 
+
+async function createEnvironment( req, res ) {
+  
+   env_secret = req.body.secret;
+  
+   const newEnvironment = {
+      id: _randomString( 20 ), 
+      name: 'On-Premises Quick-start',
+      organizationId: _randomString( 20 ),
+      accessKeys: [ { value: _randomString( 20 ) } ],
+      services: [
+         {
+            id: _randomString( 24 ), 
+            type: 'easy-image'
+         },
+         {
+            id: _randomString( 24 ),
+            type: 'collaboration'
+         }
+      ] 
+   };
+   const timestamp = Date.now();
+   const uri = `http://ckeditor-cs:8000/environments`;
+   const signature = _generateSignature( env_secret, 'POST', uri, timestamp, newEnvironment );
+   const headers = {
+      'X-CS-Signature': signature,
+      'X-CS-Timestamp': timestamp
+   };
+  
+   try {
+      await axios.post( uri, newEnvironment, { headers: headers } ) ;
+      console.log( 'New Environment created.' );
+      console.log( `EnvironmentId: ${ newEnvironment.id } AccessKey: ${ newEnvironment.accessKeys[ 0 ].value }` );
+      environment_id = newEnvironment.id;
+      access_key = newEnvironment.accessKeys[ 0 ].value;
+
+      injectEndpoints( req.body.ip, req.body.csPort, req.body.nodePort );
+
+      res.send( 'Done' );
+   } catch ( err ) {
+      console.log( err.message );
+      res.status( 500 ).send( 'Could not create environment. ' + err.message );
+   }
+}
+
+function injectEndpoints( ip, csPort, nodePort ) { 
+   const tokenEndpoint = `http://${ ip }:${ nodePort }/token`;
+   const uploadUrl = `http://${ ip }:${ csPort }/easyimage/upload`;
+   const websocketUrl = `ws://${ ip }:${ csPort }/ws`;
+  
+   let dialogJsFile = fs.readFileSync( './editor/sample/configuration-dialog/configuration-dialog.js' ).toString();
+
+   dialogJsFile = dialogJsFile.replace( /http:\/\/.*\/token/, tokenEndpoint );
+   dialogJsFile = dialogJsFile.replace( /http:\/\/.*\/easyimage\/upload/,uploadUrl );
+   dialogJsFile = dialogJsFile.replace( /ws:\/\/.*\/ws/,websocketUrl );
+
+   fs.writeFileSync( './editor/sample/configuration-dialog/configuration-dialog.js', dialogJsFile, 'utf8' );
+}
```

Comment:
It can use localhost there, but you will be able to connect only from your machine without editing endpoints. Here I wanted to use IP of the host to inject it into default endpoints, so anyone in the same network will be also able to easily connect to the document:
![image](https://user-images.githubusercontent.com/58658048/116690347-8a580400-a9b9-11eb-9f34-17585a0cee2e.png)